bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieRequestProcessor.java

```diff
@@ -430,52 +434,57 @@ public void processRequest(Object msg, Channel c) {
-                    c.writeAndFlush(response.build());
-                    if (statsEnabled) {
-                        bkStats.getOpStats(BKStats.STATS_UNKNOWN).incrementFailedOps();
-                    }
-                    break;
+            restoreMdcContextFromRequest(r);
+            try {
+                BookkeeperProtocol.BKPacketHeader header = r.getHeader();
+                switch (header.getOperation()) {
+                    case ADD_ENTRY:
+                        processAddRequestV3(r, c);
+                        break;
+                    case READ_ENTRY:
+                        processReadRequestV3(r, c);
+                        break;
+                    case FORCE_LEDGER:
+                        processForceLedgerRequestV3(r, c);
+                        break;
+                    case AUTH:
+                        LOG.info("Ignoring auth operation from client {}", c.remoteAddress());
+                        BookkeeperProtocol.AuthMessage message = BookkeeperProtocol.AuthMessage
+                                .newBuilder()
+                                .setAuthPluginName(AuthProviderFactoryFactory.AUTHENTICATION_DISABLED_PLUGIN_NAME)
+                                .setPayload(ByteString.copyFrom(AuthToken.NULL.getData()))
+                                .build();
+                        BookkeeperProtocol.Response.Builder authResponse = BookkeeperProtocol.Response
+                                .newBuilder().setHeader(r.getHeader())
+                                .setStatus(BookkeeperProtocol.StatusCode.EOK)
+                                .setAuthResponse(message);
+                        c.writeAndFlush(authResponse.build());
+                        break;
+                    case WRITE_LAC:
+                        processWriteLacRequestV3(r, c);
+                        break;
+                    case READ_LAC:
+                        processReadLacRequestV3(r, c);
+                        break;
+                    case GET_BOOKIE_INFO:
+                        processGetBookieInfoRequestV3(r, c);
+                        break;
+                    case START_TLS:
+                        processStartTLSRequestV3(r, c);
+                        break;
+                    default:
+                        LOG.info("Unknown operation type {}", header.getOperation());
+                        BookkeeperProtocol.Response.Builder response =
+                                BookkeeperProtocol.Response.newBuilder().setHeader(r.getHeader())
+                                        .setStatus(BookkeeperProtocol.StatusCode.EBADREQ);
+                        c.writeAndFlush(response.build());
+                        if (statsEnabled) {
+                            bkStats.getOpStats(BKStats.STATS_UNKNOWN).incrementFailedOps();
+                        }
+                        break;
+                }
+            } finally {
+                MDC.clear();
```

Comment:
call it conditionally? Only if the conf is enabled.

Reply:
I thought about it, but MDC is not set otherwise so clear (that checks if underlying map is not empty) vs checking for that flag are the same at the end.